unikernel.ml

```diff
@@ -379,40 +410,31 @@ struct
     Albatross_state.query stack state ~domain:user_name (`Block_cmd `Block_info)
     >|= function
     | Error msg ->
-        Logs.err (fun m ->
-            m "Error querying volumes on '%s': %s"
-              (Configuration.name_to_str state.configuration.name)
-              msg);
+        update_albatross_status state (Some (`Configuration, `String msg));
         []
-    | Ok (_hdr, `Success (`Block_devices blocks)) -> blocks
+    | Ok (_hdr, `Success (`Block_devices blocks)) ->
+        update_albatross_status state None;
+        blocks
     | Ok reply ->
-        Logs.err (fun m ->
-            m "Expected a block info reply from '%s', received %a"
-              (Configuration.name_to_str state.configuration.name)
-              (Vmm_commands.pp_wire ~verbose:false)
-              reply);
+        update_albatross_status state
+          (Some (`Transient, `Res (reply, "block info")));
         []
```

Comment:
Hmm, can't we do the albatross status updates directly in `Albatross_state.query`? There we also know all the fine details...